### PR TITLE
Implement design system for Medicaid director/office section

### PR DESCRIPTION
**Pull request by mgwalker**
Implements #1331 

### This pull request changes...
- switches from custom input types to CMS DS `<TextField>`
- switches "state" from text box to dropdown box
- removes ace-css classes

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This feature is done when...
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/containers/ApdStateProfileMedicaidOffice.js [hunk 1]
```diff
@@ -21,68 +22,83 @@ class ApdStateProfile extends Component {
 
     return (
       <Fragment>
-        <h3 className="mt0">{t(`${dirTRoot}.title`)}</h3>
-        <div className="mb3">
-          <Input
+        <h4 className="ds-h4 ds-u-margin-bottom--0">
```
Comment (mgwalker) [inline web/src/containers/ApdStateProfileMedicaidOffice.js]: Not 100% sure on the need for this 0 margin bottom.  There isn't much margin anyway, but looking at Jen's screenshots, it looked like there should be less.

File: web/src/containers/ApdStateProfileMedicaidOffice.js [hunk 2]
```diff
@@ -21,68 +22,83 @@ class ApdStateProfile extends Component {
...
-              <Input
-                name="apd-state-profile-city"
-                label={t(`${offTRoot}.labels.city`)}
-                value={medicaidOffice.city}
-                onChange={this.handleChange('medicaidOffice', 'city')}
-              />
-            </div>
-            <div className="col col-6 sm-col-3 px1">
-              <Input
+          <div className="ds-l-row">
```
Comment (mgwalker) [inline web/src/containers/ApdStateProfileMedicaidOffice.js]: This is the only way I could think of to get the two inputs on the same row.  One `TextField` is 9 columns, by default, which is where the 6 and 3 below come from.

**Comment by jeromeleecms** [comment]
@lauraponce Not sure if this is in scope for this issue, but is there any concerns with the way the state text box cuts of all the test?

image

The rest of the implementation looks very close to what was proposed in #1331 (some minor spacing issues). Will do some light accessibility testing (if it doesn't crash my computer) in a moment.

**Comment by jeromeleecms** [comment]
Tagging @nkkl for accessibility testing!

**Comment by nkkl** [comment]
Didn't find any major screen reader problems, but there's a nested (empty) text element showing up that should probably be skipped:

image

**Comment by mgwalker** [comment]
@lauraponce So it turns out I was wrong about the textboxes being 9 columns.  They're 460px, max width.  I suspect it's still doable but I'll need to pair with Heather to figure it out because I'm now thoroughly confused about what the design system is doing in this case.  😄

File: web/src/containers/ApdStateProfileMedicaidOffice.js (hunk 1 above)
Comment (hbillings) [reply web/src/containers/ApdStateProfileMedicaidOffice.js]: It occurs to me now that we could just nix the `h4` and use the `fieldset legend` as the section title. Suggested code incoming!

**Comment by jeromeleecms** [comment]
@mgwalker @lauraponce Just checking in on this -- are we still looking to adjust the text box width for the city and state fields?

**Comment by mgwalker** [comment]
After scheduling a meeting with Heather yesterday and forgetting to actually invite her to it, we have the real meeting today, to figure out how to get the spacing right.  That's the only thing holding this one up!

**Comment by jeromeleecms** [comment]
Sounds good - I didn't realize that unintentional solo meeting you mentioned yesterday was about this PR. I'll await updates!

**Comment by mgwalker** [comment]
@hbillings If you want to look at the whitespace around the fieldsets, it's ready for you.  Otherwise it's good to merge whenever.

File: web/src/styles/scss/forms.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  border: 0;
+  max-width: $input-max-width;
+}
+
+/* only legends that are first-level descendents of sections should inherit this style */
+section > fieldset {
+  legend {
+    margin-top: $spacer-3;
+    font-size: $h4-font-size;
+    font-weight: bold;
```
Comment (hbillings) [inline web/src/styles/scss/forms.scss]:
```suggestion
    float: left;
    width: 100%;
```

**Comment by mgwalker** [comment]
Here's what it looks like now:

Screen Shot 2019-04-03 at 2 10 05 PM

**Pull request merged**